# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build libpng-build sdl2-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

pkg-config --exists jpeg-build && REQUIRES="$REQUIRES jpeg-build"
pkg-config --exists libwebp-build && REQUIRES="$REQUIRES libwebp-build"
pkg-config --exists tiff-build && REQUIRES="$REQUIRES tiff-build"

# configure (ac-2.69; am-1.16; lt-2.2.6)
libtoolize -c -f; aclocal -I acinclude; autoconf; automake -a -c
PKG_CONFIG_PATH=/x11/lib/pkgconfig LDFLAGS=-Wl,-rpath,/x11/lib ./configure --disable-static --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/bin
install .libs/showimage $DESTDIR/x11/bin/showimage2
install -d $DESTDIR/x11/include/SDL2
install -m 644 SDL_image.h $DESTDIR/x11/include/SDL2
install -d $DESTDIR/x11/lib
install .libs/libSDL2_image-2.0.so.0.* $DESTDIR/x11/lib/libSDL2_image-2.0.so.0
ln -sf libSDL2_image-2.0.so.0 $DESTDIR/x11/lib/libSDL2_image.so
install -d $DESTDIR/x11/lib/pkgconfig
install -m 644 SDL2_image.pc $DESTDIR/x11/lib/pkgconfig

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/sdl2_image-build.pc << EOF
Name: SDL2_image
Version: 2.0.4
Description: Image loading library for SDL
Requires: $REQUIRES

devel=\\
/x11/include/SDL2/SDL_image.h \\
/x11/lib/libSDL2_image.so \\
/x11/lib/pkgconfig/SDL2_image.pc

exec=\\
/x11/bin/showimage2 \\
/x11/lib/libSDL2_image-2.0.so.0
EOF

exit

# patch
--- SDL2_image-2.0.4.orig/configure.in
+++ SDL2_image-2.0.4/configure.in
@@ -1,5 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(README.txt)
+AC_INIT([SDL2_image], [2.0.4])
+AC_CONFIG_SRCDIR([README.txt])
+AC_CONFIG_MACRO_DIR([acinclude])
 
 dnl Set various version strings - taken gratefully from the GTk sources
 
@@ -42,7 +44,7 @@
 AC_CANONICAL_HOST
 
 dnl Setup for automake
-AM_INIT_AUTOMAKE(SDL2_image, $VERSION)
+AM_INIT_AUTOMAKE([foreign])
 
 dnl Check for tools
 AC_PROG_LIBTOOL

# source
https://www.libsdl.org/projects/SDL_image/release/SDL2_image-2.0.4.tar.gz
